### ISSUE TITLE
Create a state machine graph of the workflow for a given object

### ISSUE THREAD

**Issue by mjgiarlo** [projecthydra-labs/hyrax#226]
<a href="https://github.com/jeremyf"><img src="https://avatars.githubusercontent.com/u/2130?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jeremyf](https://github.com/jeremyf)**
_Friday Sep 23, 2016 at 15:15 GMT_
_Originally opened as https://github.com/projecthydra/curation_concerns/issues/1030_

----

In [Sipity](https://github.com/ndlib/sipity), we are able to generate a state machine diagram using [DOT notation](http://graphviz.org). This is a helpful debugging tool for developers

Port the state machine diagram generation from Sipity into Curation Concerns.

**Comment by mjgiarlo** [comment]
@vantuyls Do you want to slot this into MD milestone 2 or 3, or leave in its current milestone?

**Comment by vantuyls** [comment]
It isn't entirely clear to me what this does for the user/administrator. Can @mjgiarlo or @jeremyf eli5?

**Comment by jeremyf** [comment]
@vantuyls The goal of the graph is to be able to provide a visual representation of the workflow…what actions move to what state, what roles are assigned, etc.

**Comment by jcoyne** [comment]
Port https://github.com/samvera-labs/hyku/pull/1314 to hyrax?

**Comment by vantuyls** [comment]
yes please.
